internal/liveness/metrics.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2019 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package liveness
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/ceph/ceph-csi/internal/util"
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+var (
+	rbdMapping = *prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "csi",
+		Name:      "ceph_rbd",
+		Help:      "ceph rbd image pvc mapping"},
+		[]string{"pvc_name", "pvc_namespace", "pv_name", "pool", "imageID"})
+	cephfsMapping = *prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: "csi",
+		Name:      "ceph_fs",
+		Help:      "cephfs subvolume pvc mapping"},
+		[]string{"pvc_name", "pvc_namespace", "pv_name", "volume_name", "subvolume_group", "subvolume_name"})
+)
+
+func getMapping(timeout time.Duration) {
+	klog.V(4).Infof("Start recording mapping")
+	client := util.NewK8sClient()
+	pvcs, err := client.CoreV1().PersistentVolumeClaims("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		klog.Errorf("Get pvclist failed: %v", err)
+		return
+	}
+	for _, pvc := range pvcs.Items {
+		pvcName := pvc.Name
+		pvcNameSpace := pvc.Namespace
+		pvName := pvc.Spec.VolumeName
+		pvInfo, err := client.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+		if err != nil {
+			klog.V(4).Infof("Get pvinfo failed: %v", err)
+			continue
+		}
+		if pvInfo.Spec.CSI != nil {
+			volumeHandle := pvInfo.Spec.CSI.VolumeHandle
+			volumeHandleSlice := strings.Split(volumeHandle, "-")
+			index := len(volumeHandleSlice) - 5
+			imageName := strings.Join(append([]string{"csi-vol"}, volumeHandleSlice[index:]...), "-")
```

Comment:
csi-vol can be overridden with storageclass argument, created an issue to add these parameters to storageclass.yaml https://github.com/ceph/ceph-csi/issues/1087

Reply:
so we can get volumeNamePrefix info from storageclass

Reply:
done